demos/MLSys/demo.ipynb

```diff
@@ -0,0 +1,125 @@
+      "text/plain": [
+       "<drawSvg.raster.Raster at 0x7f84d868e310>"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "grid_width = 10\n",
+    "d = draw.Drawing(1600, 900, origin = (0, 0))\n",
+    "for x in range(output_dims[0]):\n",
+    "    for y in range(output_dims[1]):\n",
+    "        d.append(draw.Rectangle(100 * x + grid_width, 100 * y + grid_width, \n",
+    "                                100, 100, \n",
+    "                        stroke_width = grid_width, stroke = 'black',\n",
+    "                        fill = '#aaddff'))\n",
+    "d.rasterize()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "binder = plaidml.exec.Binder(program)\n",
+    "executable = binder.compile()\n",
+    "executable.run()"
```

Comment:
I'm wondering if we really want to run anything, it will probably go very slowly depending on what code is submitted, so generally we should probably disallow it...

Once we have the compiler API, we can use that to retrieve the output of various passes.

Reply:
Oh, for this demo we need that, OK :)